common/test-setting-immutable-prototype.js

```diff
@@ -1,24 +1,27 @@
-self.testSettingImmutablePrototypeToNewValueOnly = (prefix, target, newValue, newValueString) => {
+self.testSettingImmutablePrototypeToNewValueOnly =
+  (prefix, target, newValue, newValueString, { isSameOriginDomain }) => {
   test(() => {
     assert_throws(new TypeError, () => {
       Object.setPrototypeOf(target, newValue);
     });
   }, `${prefix}: setting the prototype to ${newValueString} via Object.setPrototypeOf should throw a TypeError`);
 
+  const dunderProtoError = isSameOriginDomain ? { name: "TypeError" } : { name: "SecurityError" };
+
   test(() => {
-    assert_throws(new TypeError, function() {
+    assert_throws(dunderProtoError, function() {
       target.__proto__ = newValue;
     });
-  }, `${prefix}: setting the prototype to ${newValueString} via __proto__ should throw a TypeError`);
+  }, `${prefix}: setting the prototype to ${newValueString} via __proto__ should throw a ${dunderProtoError.name}`);
```

Comment:
I guess that makes this last bit harder, but we could just stash the string we care about in another variable to use here.